testproject/Assets/Scripts/SyncTransform.cs

```diff
@@ -12,108 +13,141 @@ namespace MLAPI
             }
         }
 
-        private void Start()
+        void Start()
         {
-            m_NetVarPos.Settings.WritePermission = NetworkedVarPermission.Everyone;
-            m_NetVarRot.Settings.WritePermission = NetworkedVarPermission.Everyone;
+            m_varPos.Settings.WritePermission = NetworkedVarPermission.Everyone;
+            m_varRot.Settings.WritePermission = NetworkedVarPermission.Everyone;
         }
 
-        private void FixedUpdate()
+        void FixedUpdate()
         {
-            float now = Time.fixedTime;
-            if (m_LastSent == 0.0f)
+            float now = Time.time;
+            if (m_lastSent == 0.0f)
             {
-                m_LastSent = now;
+                m_lastSent = now;
             }
 
             // if this.gameObject is local let's send its position
             if (IsLocalPlayer)
             {
-                while ((now - m_LastSent) > k_UpdateRate)
-                {
-                    m_LastSent += k_UpdateRate;
-
-                    m_NetVarPos.Value = gameObject.transform.position;
-                    m_NetVarRot.Value = gameObject.transform.rotation;
-                }
+                m_varPos.Value = gameObject.transform.position;
+                m_varRot.Value = gameObject.transform.rotation;
             }
             else
             {
-                // todo: do we want to perform local interpolation on Update() instead?
+                if (!interpolate)
+                {
+                    return;
+                }
 
                 // let's interpolate the last received transform
                 if (m_PosTimes[0] >= 0.0 && m_PosTimes[1] >= 0.0)
                 {
-                    if (m_PosTimes[1] - m_PosTimes[0] > k_Epsilon)
+                    var before = gameObject.transform.position;
+
+                    if (m_PosTimes[1] - m_PosTimes[0] > k_epsilon)
+                    {
+                        if ((now - m_PosTimes[0]) / (m_PosTimes[1] - m_PosTimes[0]) < 2.0)
+                        {
+                            gameObject.transform.position = Vector3.LerpUnclamped(
+                                m_PosStore[0],
+                                m_PosStore[1],
+                                (now - m_PosTimes[0]) / (m_PosTimes[1] - m_PosTimes[0]));
+                        }
+                    }
+                    else
                     {
-                        gameObject.transform.position = Vector3.LerpUnclamped(
-                            m_PosStore[0],
-                            m_PosStore[1],
-                            (now - m_PosTimes[0]) / (m_PosTimes[1] - m_PosTimes[0]));
+                        gameObject.transform.position = m_PosStore[1];
                     }
+
+                    var after = gameObject.transform.position;
+
+                    //Debug.Log("[3] Updated position from " + before + " to " + after);
```

Comment:
remove this line entirely?

Reply:
I'd keep it for now, it's need until we resolve the quaternion networked var issue. If this was mlapi per-se, yes, but since this is testproject, I think it can be there a little while